src/state-manager.ts

```diff
@@ -0,0 +1,85 @@
+import * as vscode from "vscode";
```

Comment:
this is also going to be a main logic file so consider adding tests to it.
I know now the logic is really "simple & stupid" but still (not a priority)

Reply:
fcdbf0ff3057ded31d46c33ae0c91d502de717e1